src/foam/u2/crunch/CrunchController.js

```diff
@@ -243,6 +264,50 @@ foam.CLASS({
           functionData: capabilitiesSections,
           capabilityId: capabilityId
         })).end();
+    },
+    function save(wizardlet) {
+      var isAssociationCapability = foam.nanos.crunch.AssociationCapability.isInstance(wizardlet.capability);
+      var associatedEntity = isAssociationCapability ? this.subject.realUser : 
+      wizardlet.capability.associatedEntity === 'user' ? this.subject.user : this.subject.realUser;
+
+      return this.updateUCJ(wizardlet, associatedEntity).then(() => {
+        var ucj = wizardlet.ucj;
+        if ( ucj === null ) {
+          ucj = isAssociationCapability ? 
```

Comment:
isAssociationCapability means we are saving cap to ucj on realUser ... does that really mean this will be an agentCapabilityJunction?

Reply:
associatedEntity determines whether we should save to user/realuser. is associationcapability is a special case of capability where we want to grant to "realuser when acting as user", which uses agentcapabilityjunction instead of usercapabilityjunction